crates/client/data-availability/src/lib.rs

```diff
@@ -4,23 +4,26 @@ pub mod ethereum;
 mod sharp;
 pub mod utils;
 
-use std::collections::HashMap;
 use std::marker::PhantomData;
 use std::sync::Arc;
 
 use anyhow::Result;
 use async_trait::async_trait;
 use ethers::types::{I256, U256};
+use futures::channel::mpsc;
 use futures::StreamExt;
-use mp_storage::{SN_NONCE_PREFIX, SN_STORAGE_PREFIX};
+use mc_commitment_state_diff::BlockDAData;
+use mp_hashers::HasherT;
+use pallet_starknet_runtime_api::StarknetRuntimeApi;
 use sc_client_api::client::BlockchainEvents;
 use serde::Deserialize;
 use sp_api::ProvideRuntimeApi;
-use sp_runtime::traits::Block as BlockT;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::traits::{Block as BlockT, Header};
+use utils::state_diff_to_calldata;
 
-pub type StorageWrites<'a> = Vec<(&'a [u8], &'a [u8])>;
-
-pub struct DataAvailabilityWorker<B, C>(PhantomData<(B, C)>);
+pub struct DataAvailabilityWorker<B, C, H>(PhantomData<(B, C, H)>);
+pub struct DataAvailabilityWorkerProving<B>(PhantomData<B>);
 
 #[derive(Debug, Copy, Clone, PartialEq, clap::ValueEnum)]
```

Comment:
`clap::ValueEnum`

clap should be an optional dep, and this should be behind a feature flag